provider/pkg/provider/yaml/v2/configGroup.go

```diff
@@ -0,0 +1,97 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pulumi/pulumi-kubernetes/provider/v4/pkg/clients"
+	providerresource "github.com/pulumi/pulumi-kubernetes/provider/v4/pkg/provider/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/internals"
+	pulumiprovider "github.com/pulumi/pulumi/sdk/v3/go/pulumi/provider"
+)
+
+type ConfigGroupProvider struct {
+	clientSet *clients.DynamicClientSet
+}
+
+type ConfigGroupArgs struct {
+	Files          pulumi.StringArrayInput `pulumi:"files"`
+	YAML           pulumi.StringArrayInput `pulumi:"yaml,optional"`
+	Objects        pulumi.MapArrayInput    `pulumi:"objs,optional"`
+	ResourcePrefix pulumi.StringInput      `pulumi:"resourcePrefix,optional"`
+	SkipAwait      pulumi.BoolInput        `pulumi:"skipAwait,optional"`
+}
+
+type ConfigGroupState struct {
+	pulumi.ResourceState
+	Resources pulumi.ArrayOutput `pulumi:"resources"`
+}
+
+var _ providerresource.ResourceProvider = &ConfigGroupProvider{}
+
+func NewConfigGroupProvider(opts *providerresource.ResourceProviderOptions) providerresource.ResourceProvider {
+	return &ConfigGroupProvider{
+		clientSet: opts.ClientSet,
+	}
+}
+
+func (k *ConfigGroupProvider) Construct(ctx *pulumi.Context, typ, name string, inputs pulumiprovider.ConstructInputs, options pulumi.ResourceOption) (*pulumiprovider.ConstructResult, error) {
+	comp := &ConfigGroupState{}
+	err := ctx.RegisterComponentResource(typ, name, comp, options)
+	if err != nil {
+		return nil, err
+	}
+
+	args := &ConfigGroupArgs{}
+	if err := inputs.CopyTo(args); err != nil {
+		return nil, fmt.Errorf("setting args: %w", err)
+	}
+
+	// Check if all the required args have resolved, and print a warning if not.
+	result, err := internals.UnsafeAwaitOutput(ctx.Context(), pulumi.All(
+		args.Files, args.YAML, args.Objects, args.ResourcePrefix))
+	if err != nil {
+		return nil, err
+	}
+	if !result.Known {
+		msg := fmt.Sprintf("%s:%s -- Required input properties have unknown values. Preview is incomplete.\n", typ, name)
+		_ = ctx.Log.Warn(msg, nil)
+	}
+
+	// Parse the manifests and register the resources.
+
+	comp.Resources = pulumi.All(args.Files, args.YAML, args.Objects, args.ResourcePrefix).ApplyTWithContext(ctx.Context(), func(_ context.Context, args []any) (pulumi.ArrayOutput, error) {
+		files, _ := args[0].([]string)
+		yaml, _ := args[1].([]string)
+		objects, _ := args[2].([]map[string]interface{})
+		resourcePrefix, _ := args[3].(string)
```

Comment:
If you're certain that the type assertions are valid, it would be better to use the single-return form that will panic on error. Otherwise, the assertions should be checked.

Reply:
I think the code is correct because we want the zero value as opposed to a panic when the element is `nil`. For example, we want files to be `nil` when not specified.  The type assertions are known to be valid.

Reply:
Sounds good. I'd suggest adding a comment indicating that it's done on purpose and noting the zero-value behavior.